app/gui/view/debug_scene/cloud-dashboard/cloud-view/src/id.rs

```diff
@@ -0,0 +1,302 @@
+    }
+}
+
+impl<T> FromStr for Id<T>
+where T: IdVariant
+{
+    type Err = crate::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let ksuid_str = match s.split_twice(T::PREFIX, ID_PREFIX_SEPARATOR) {
+            Some(("", "", s)) => s,
+            _ => Err(format!("\"{s}\" is not an {ID_STRUCT_NAME}."))?,
+        };
+        let ksuid = svix_ksuid::Ksuid::from_str(ksuid_str)
+            .map_err(|_| format!("\"{ksuid_str}\" is not a Ksuid."))?;
+        let marker = PhantomData;
+        Ok(Self { ksuid, marker })
+    }
+}
+
+/// This implementation is provided manually because [`Id`] is a dash-separated string, which
+/// [`serde`] can't derive [`Deserialize`] for. By manually implementing [`Deserialize`], we can
+/// delegate deserialization to our [`FromStr`] implementation which can parse an [`Id`] from a
+/// [`String`] by splitting it into its [`PREFIX`] and [`Ksuid`] components.
+/// 
+/// [`Deserialize`]: ::serde::de::Deserialize
+/// [`PREFIX`]: crate::id::IdVariant::PREFIX
+/// [`Ksuid`]: ::svix_ksuid::Ksuid
+impl<'a, T> serde::Deserialize<'a> for Id<T>
+where Id<T>: FromStr
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where D: serde::Deserializer<'a> {
+        struct IdVisitor<T>(PhantomData<T>);
+
+        impl<'a, T> de::Visitor<'a> for IdVisitor<T>
+        where Id<T>: FromStr
+        {
+            type Value = Id<T>;
+
+            fn expecting(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                f.write_str("identifier")
+            }
+
+            fn visit_newtype_struct<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+            where D: serde::Deserializer<'a> {
+                deserializer.deserialize_any(IdVisitor(PhantomData))
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where E: de::Error {
+                v.parse().map_err(|_| {
+                    let unexpected = de::Unexpected::Str(v);
+                    de::Error::invalid_value(unexpected, &"identifier string")
+                })
+            }
+        }
+
+        deserializer.deserialize_any(IdVisitor(PhantomData))
+    }
+}
+
+impl<T> Debug for Id<T>
+where T: IdVariant
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_tuple(ID_STRUCT_NAME).field(&format_args!("{}", self)).finish()
+    }
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod test {
+    use enso_prelude::*;
+
+    use crate::id;
+
+    #[test]
+    fn test_id_from_str() {
+        fn check(id_str: &str, expected_str: &str) {
+            let result = id::OrganizationId::from_str(id_str);
+            let result = result.map(|v| v.to_string());
+            let result = result.map_err(|e| e.to_string());
+            let debug_str = format!("{:?}", result);
+            assert_eq!(debug_str, expected_str);
```

Comment:
I don't like asserting on the debug strings, the format is not stable and should not be relied upon.
Ideally, you should compare actual values. When not feasible, rely on `Display`.

Reply:
Both the `Result::Ok` and `Result::Err` variant are turned into strings using `Display` here. The `Debug` comparison is only because `Result` itself doesn't implement `Display`. So we're not *really* relying on a `Debug` format here.